docs/content/why-temporal.md

```diff
@@ -0,0 +1,168 @@
+
+1. The most obvious one is that the application state machine of the customer's state quickly becomes extremely complicated.
+   For example, if a credit card charge attempt fails or sending an email fails due to a downstream system's unavailability, the state is now in limbo.
+2. Failed calls likely need to be retried for a long time, and these calls need to be throttled to not overload external resources.
+3. There needs to be logic to handle corrupted customer records to avoid blocking the whole process.
+4. Additionally, databases have performance and scalability limitations (eventually requiring sharding) and are not efficient for scenarios that require constant polling.
+
+**Queue system design approach**
+
+The next commonly employed approach is to use a timer service and queues.
+Updates are pushed to a queue while a service consumes them one at a time, updating a database, and possibly pushing more messages into other downstream queues.
+A timer service can be used to schedule queue polling or database actions.
+
+While this approach has shown to scale a bit better, the programming model can become very complex and error-prone, as there are usually no transactional updates between a queuing system, a timer service, and a database.
+
+**Temporal design approach**
+
+The Temporal Platform approach aims to encapsulate and implement the entire business logic in a simple function or object method.
+Thanks to the Temporal Platform, the function/method is durably stateful, and the implementer doesn't need to employ any additional systems to ensure consistency and fault tolerance.
+
+Here are example Workflow Definitions that implement the subscription management use case in Java, Go, and PHP:
+
+<Tabs
+defaultValue="java"
+values={[
+{label: 'Go', value: 'go'},
+{label: 'Java', value: 'java'},
+{label: 'Typescript', value: 'ts'},
+{label: 'PHP', value: 'php'},
+]
+}>
+
+<TabItem value="go">
+
+<!--SNIPSTART subscription-go-workflow-definition-->
+<!--SNIPEND-->
+
+</TabItem>
+<TabItem value="java">
+
+<!--SNIPSTART subscription-java-workflow-definition-implementation-->
+<!--SNIPEND-->
+
+</TabItem>
+<TabItem value="ts">
+
+<!--SNIPSTART subscription-ts-workflow-definition-->
+<!--SNIPEND-->
+
+</TabItem>
+<TabItem value="php">
+
+<!--SNIPSTART subscription-php-workflow-definition-implementation-->
+<!--SNIPEND-->
+
+</TabItem>
+</Tabs>
+
+Again, it is important to note that this is working application code that directly implements the business logic.
+If any of the operations take a long time, the code is not going to change.
+
+It is completely okay to be blocked on `chargeCustomerForBillingPeriod` for a day or more if the downstream processing service is down or not responding.
+In the same way, it is a completely normal operation to sleep for 30 days directly inside the Workflow code.
+This is possible because infrastructure failures are not going to affect the Workflow state – including threads, blocking calls, and any variables.
```

Comment:
```suggestion
This is possible because infrastructure failures won't affect the Workflow state—including threads, blocking calls, and any variables.
```